mdata/notifierKafka/notifierKafka.go

```diff
@@ -69,22 +69,77 @@ func (c *NotifierKafka) start() {
 	pre := time.Now()
 	processBacklog := new(sync.WaitGroup)
 	for _, partition := range partitions {
-		var offsetTime int64
+		type offset struct {
+			offsetStart int64
+			offsetTime  int64
+			offsetName  string
+			offsetError error
+		}
+		var startOffset int64
+		var validOffset bool
+		var offsetFromDuration int64
+
+		offsets := make([]offset, 0, 3)
+
+		if offsetStr != "oldest" && offsetStr != "newest" {
+			offsetFromDuration = time.Now().Add(-1*offsetDuration).UnixNano() / int64(time.Millisecond)
+		}
+
+		offsets[0].offsetTime = sarama.OffsetOldest
+		offsets[0].offsetName = "oldest"
+		offsets[1].offsetTime = sarama.OffsetNewest
+		offsets[1].offsetName = "newest"
+		offsets[2].offsetTime = offsetFromDuration
+		offsets[2].offsetName = "custom"
+
+		// get all of the offsets
+		for i := 0; i < 2; i++ {
```

Comment:
this would be cleaner as
```
for i := range offsets {
```

Reply:
Yeah. The code is still a little sloppy right now, I was just trying to get something up so people could see what I was trying to do. I will work on refactoring it a little bit.